ksqldb-functional-tests/src/test/resources/historical_plans/url_-_extract_a_protocol_from_a_URL_using_URL_EXTRACT_PROTOCOL_-_PROTOBUFs_-_PROTOBUF_NOSR/7.4.0_1668502778749/plan.json

```diff
@@ -0,0 +1,214 @@
+{
+  "plan" : [ {
+    "@type" : "ksqlPlanV1",
+    "statementText" : "CREATE STREAM TEST (K STRING KEY, URL STRING) WITH (KAFKA_TOPIC='test_topic', KEY_FORMAT='KAFKA', VALUE_FORMAT='PROTOBUF_NOSR');",
+    "ddlCommand" : {
+      "@type" : "createStreamV1",
+      "sourceName" : "TEST",
+      "schema" : "`K` STRING KEY, `URL` STRING",
+      "timestampColumn" : null,
+      "topicName" : "test_topic",
+      "formats" : {
+        "keyFormat" : {
+          "format" : "KAFKA",
+          "properties" : { }
+        },
+        "valueFormat" : {
+          "format" : "PROTOBUF_NOSR",
+          "properties" : {
+            "unwrapPrimitives" : "true"
```

Comment:
The only difference between these new historic plans and the existing ones should be the addition of this `unwrapPrimitives` property for all `PROTOBUF_NOSR` usages, right? 

Can you also confirm my understanding regarding changes to `unwrapPrimitives` and `PROTOBUF_NOSR` in this PR: it looks like `PROTOBUF_NOSR` was already set up to handle the config option but `unwrapPrimitives` was never being set, which means we were always using `unwrapPrimitives=false`. Then after the changes in this PR, we'll start setting `unwrapPrimitives=true` for all new queries, but existing queries will continue to run with `unwrapPrimitives=false`, which means the PR maintains backwards compatibility. Is this true? Might be worth running a quick manual test to ensure that we're not breaking backwards compatibility here. AFAICT from the code changes we should be fine. 

Normally I'd feel fine about just the QTT coverage but something weird seems to be going on with (at least some of) the existing QTT coverage. For example, this QTT that I'm commenting on looks like it isn't even running `PROTOBUF_NOSR` (despite what the name suggests) in its current version (https://github.com/confluentinc/ksql/blob/master/ksqldb-functional-tests/src/test/resources/historical_plans/url_-_extract_a_protocol_from_a_URL_using_URL_EXTRACT_PROTOCOL_-_PROTOBUF/7.1.0_1633126034008/plan.json), which means it's not offering any compatibility guarantees for the `PROTOBUF_NOSR` upgrade that we're interested in.

Reply:
Hmmm very good point. I think I realized in the end that things were fine as they were for `unwrapPrimitives` - it only has an effect if I infer an SQL schema from an existing protobuf schema. Without using a schema registry, it is not possible to define an "existing protobuf schema". So whether `unwrapPrimitives` is enabled or not does not make a difference. Sorry for mixing this into the PR. I reverted all changes that touched `unwrapPrimitives`.